ledger-core/virtual/integration/method_test.go

```diff
@@ -1232,3 +1230,237 @@ func TestVirtual_CallMultipleContractsFromContract_Ordered(t *testing.T) {
 
 	mc.Finish()
 }
+
+// call A.Foo -> B.Bar two times
+func TestVirtual_CallContractTwoTimes(t *testing.T) {
+	t.Log("C5183")
```

Comment:
Let's concentrate on testing Sequence field a bit. I think it's ok to drop passing requests to B object, e.g. avoid resending messages from A to B and generate response manually. This will cut test in half. Instead we can add more checks around sequence, for example do the following calls:

- call to A.Foo
-- A.Foo calls B.Some twice
- second call to A.Foo
-- A.Foo calls B.Some twice

check that Sequence on outgoing calls is 1, 2, 1, 2.

What do you think?

Reply:
done